core/drivers/crypto/caam/utils/utils_dmaobj.c

```diff
@@ -915,8 +921,14 @@ size_t caam_dmaobj_copy_ltrim_to_orig(struct caamdmaobj *obj)
 	}
 
 do_copy:
-	if (off < obj->orig.length)
-		dst_rlen = obj->orig.length - off;
+	/*
+	 * The maximum data size to copy cannot exceed the output buffer size
+	 * (obj->orig.length) and cannot exceed the data processed by the
+	 * CAAM (obj->sgtbuf.length).
+	 */
+	dst_rlen = MIN(obj->orig.length, obj->sgtbuf.length);
```

Comment:
this changes `dst_rlen` value. I think test at line 938 is corrupted by this change.

Reply:
What do you mean by corrupted? To me, this test still looks valid

Reply:
With previous implementation `dst_rlen` was 0 when  `off >= obj->orig.length`. Here it likely always gets non 0 value. It could cause trouble IMO.

Reply:
You are right, the change might have an impact on the following `if (!dst_rlen) {` statement.
In the end, this change in the function `caam_dmaobj_copy_ltrim_to_orig()` is not needed. This function is only used in the RSA crypto operation and the `obj->orig.length` cannot be smaller than `obj->sgtbuf.length` because of the SGT allocation for the RSA. Thus, the modification done is not needed.

Reply:
@etienne-lms Gentle reminder!